test/functional/mongo_client_options.test.js

```diff
@@ -96,6 +97,17 @@ describe('MongoClient Options', function() {
     }
   });
 
+  it('should default socketTimeout to infinity', function(done) {
+    const client = new MongoClient(this.configuration.url());
+    client.connect(() => {
+      expect(client.s.options.socketTimeoutMS).to.deep.equal(0);
+      for (const connection of client.topology.s.coreTopology.connections()) {
+        expect(connection.socketTimeout).to.deep.equal(0);
+      }
+      done();
+    });
+  });
+
```

Comment:
@mbroadst I saw your comment about missing testing on the 4.0 version of this, added a test here that I'll also add to 4.0 if this is all good ✅

Reply:
LGTM!